csFastFloat/Constants/Constants.cs

```diff
@@ -19,7 +22,32 @@ internal static class Constants
     internal const int smallest_power_of_five = -342;
     internal const int largest_power_of_five = 308;
 
-    internal readonly static ulong[] power_of_five_128 = {
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    internal static byte get_powers(uint n)
+    {
+      Debug.Assert(n < powersTable.Length);
+      ref byte tableRef = ref MemoryMarshal.GetReference(powersTable);
+      return Unsafe.AddByteOffset(ref tableRef, (IntPtr)n);
+    }
+
+    private static ReadOnlySpan<byte> powersTable => new byte[19] {
+      0,  3,  6,  9,  13, 16, 19, 23, 26, 29, //
+      33, 36, 39, 43, 46, 49, 53, 56, 59,     //
+    };
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    internal static ulong get_power_of_five_128(int index)
+    {
+#if NET5_0
+      Debug.Assert(index < power_of_five_128.Length);
```

Comment:
This would only kick in debug mode, right?

Reply:
Yep, it's removed in Release-builds by the compiler.
Here it's just as safety-net to get a crash in debug-build if something went wrong and would access outside of bounds.